dashboard/app/views/levels/_teacher.html.haml

```diff
@@ -1,4 +1,4 @@
-- if current_user.try(:teacher?) && @script_level
+- if current_user.try(:teacher?)
```

Comment:
do we know what `&& @script_level` was protecting us from previously? I don't see the teacher panel show up on urls like `/levels/11123`, but there could be another scenario I'm missing.

Reply:
also, there are a few places below where `@script_level` is accessed without performing a nil check first. please add nil checks on those (e.g. `@script_level&.`...). 

To go a bit further than my previous comment, I think it would be worth understanding when `@script_level` would be nil and making sure the right thing is happening in that case before proceeding. Alternately/additionally, you could change this check to be something like `(@script_level || is_stage_extra)` to catch the right set of conditions.

Reply:
you're right, great catch. previously, this partial (`_teacher.html.haml`) was only used in the [`levels/show` view](https://github.com/code-dot-org/code-dot-org/blob/c584af78f697a53b0916fb3b0fff7a63bd02ab0e/dashboard/app/views/levels/show.html.haml#L69), which is invoked by these controller actions:
1. [`levels/show`](https://github.com/code-dot-org/code-dot-org/blob/c584af78f697a53b0916fb3b0fff7a63bd02ab0e/dashboard/app/controllers/levels_controller.rb#L28-L39) (the one you mentioned)
2. [`levels#embed_level`](https://github.com/code-dot-org/code-dot-org/blob/c584af78f697a53b0916fb3b0fff7a63bd02ab0e/dashboard/app/controllers/levels_controller.rb#L226-L238)
3. [`script_levels#show`](https://github.com/code-dot-org/code-dot-org/blob/c584af78f697a53b0916fb3b0fff7a63bd02ab0e/dashboard/app/controllers/script_levels_controller.rb#L75-L138)
4. [`script_levels#stage_extras`](https://github.com/code-dot-org/code-dot-org/blob/c584af78f697a53b0916fb3b0fff7a63bd02ab0e/dashboard/app/controllers/script_levels_controller.rb#L175-L202)

we do not want the teacher panel to display in cases 1 & 2, so i need to refactor this line to something like:
```
- if current_user.try(:teacher?) && (@script_level || @stage_extras)
```

Reply:
looks like we commented at the same time with similar solutions :) sounds like a great plan, thanks for tracking these all down!